graph/src/components/ethereum/adapter.rs

```diff
@@ -946,12 +946,17 @@ pub fn blocks_with_triggers(
     Box::new(
         trigger_futs
             .concat2()
-            .join(adapter.clone().block_hash_by_block_number(
-                &logger,
-                chain_store.clone(),
-                to,
-                true,
-            ))
+            .join(
+                adapter
+                    .clone()
+                    .block_hash_by_block_number(&logger, chain_store.clone(), to, true)
+                    .then(move |to_hash| match to_hash {
+                        Ok(n) => {
+                            n.ok_or_else(|| format_err!("Block {} not found in the chain", to))
```

Comment:
Is there any way we can include the Ethereum provider in the error message? Is that info available here?

Reply:
I agree that would be nice. I wouldn't dump the whole URL though for security reasons (e.g. basic auth included in the URL) I'd do only the host perhaps.